runners/runner-local/src/error.rs

```diff
@@ -1,22 +1,47 @@
-use error_stack::Context;
+use error_stack::{report, Context, Report, ResultExt};
 use std::fmt;
 
 #[derive(Debug)]
 pub enum LocalRunnerError {
-    Internal,
+    Internal(String),
     MissingArgument(String),
-    InvalidArgument { argument: String, reason: String },
+    InvalidArgument(String, String),
     NotFound(String),
     AlreadyExists(String),
 }
 impl LocalRunnerError {
+    pub fn internal(message: &str) -> Report<LocalRunnerError> {
+        report!(LocalRunnerError::Internal(message.to_string()))
+            .attach_printable(message.to_string())
+    }
+    pub fn missing_argument(argument: &str) -> Report<LocalRunnerError> {
+        report!(LocalRunnerError::MissingArgument(argument.to_string()))
+            .attach_printable(format!("missing required argument: {argument}"))
+    }
+    pub fn invalid_argument(argument: &str, reason: &str) -> Report<LocalRunnerError> {
+        report!(LocalRunnerError::InvalidArgument(
+            argument.to_string(),
+            reason.to_string()
+        ))
+        .attach_printable(format!("invalid argument {argument}: {reason}"))
+    }
+    pub fn not_found(name: &str) -> Report<LocalRunnerError> {
+        report!(LocalRunnerError::NotFound(name.to_string()))
+            .attach_printable(format!("{} not found", name))
+    }
+    pub fn already_exists(name: &str) -> Report<LocalRunnerError> {
+        report!(LocalRunnerError::AlreadyExists(name.to_string()))
+            .attach_printable(format!("{} already exists", name))
+    }
+
     pub fn to_tonic_status(&self) -> tonic::Status {
         match self {
-            LocalRunnerError::Internal => tonic::Status::internal("internal error"),
+            // TODO: include the internal error message in the tonic message but only for local runner
```

Comment:
What if we simply print it as warning? `warn!(message = message, "runner error");`

Reply:
We're already printing it as a warning before we call `to_tonic_status`.

This comment is here because, we are already giving up too much information in the status value and we're trying to be discrete for internal errors